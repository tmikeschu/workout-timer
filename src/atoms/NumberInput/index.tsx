import * as React from "react";
import { styled } from "theme";

const Container = styled.input`
  border: none;
  background: transparent;
  color: ${({ theme }): string => theme.body};
  font-size: 4rem;
  /* Chrome, Safari, Edge, Opera */
  &::-webkit-outer-spin-button,
  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  /* Firefox */
  &[type="number"] {
    -moz-appearance: textfield;
  }
`;

const NumberInput: React.FC<React.HTMLProps<HTMLInputElement>> = props => {
  return <Container type="number" {...props} />;
};

export default NumberInput;
