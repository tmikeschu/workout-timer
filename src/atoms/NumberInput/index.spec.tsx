import * as React from "react";
import { render } from "@testing-library/react";
import NumberInput from "./";

describe("NumberInput", () => {
  it("renders correctly", () => {
    const tree = render(<NumberInput />).container;
    expect(tree).toMatchSnapshot();
  });
});
