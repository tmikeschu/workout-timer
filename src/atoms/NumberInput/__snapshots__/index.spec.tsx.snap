// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders correctly 1`] = `
<div>
  <input
    class="css-7pvvo7"
    type="number"
  />
</div>
`;
